apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -1008,6 +1008,10 @@ export function getSectionRows(state, sectionIds) {
   }));
 }
 
+export function getAssignmentName(state, sectionId) {
+  const {sections, validAssignments} = getRoot(state);
+  return assignmentNames(validAssignments, sections[sectionId])[0];
```

Comment:
I read the text of the assignmentNames function, it points to the assignmentsForSection function which I have a couple of concerns about.

It looks like we can get assignments from either sections having courseIds, or sections having scriptIds. What is the difference between an assignment gotten from a courseId, verses from a scriptId? Are there situations where you might have one and not the other? What are the situations where a section wouldn't have either?

When using functions from our redux codebase, all of the ambiguities that arise from that code propogate down the line into whatever diff is using that function. Therefore, we can't really be sure that we've fixed the issue fully unless we understand all of these cases. Do you know the answer to these questions? If not, my reccomendation is to dig into it.

Reply:
I understand all of the cases. Course and scripts are different. For example CSP 2019 is a course, and CSP Unit 1 2019 is a script. 

A section could have neither a courseId or a scriptId if nothing is assigned to the section. 
A section could have only a courseId if a full course is assigned to it, but not any of the units within that course. 
A section could have only a scriptId if it is assigned a script that is not part of a course, eg. a n Hour of Code script. 
A section could have both a scriptId and a courseId if it is assigned a unit in a full course.

Reply:
Thanks! Now I also understand these cases 🙂

I think it'd be a good idea to document this in the code - unit tests for these cases would be an ideal way to do that